dl4j-labs/src/main/java/ai/certifai/training/classification/OverfittingGermanCreditClassification.java

```diff
@@ -0,0 +1,372 @@
+ *
+ * Search for the keyword "SOLUTION" to see what can be used to avoid overfitting
+ *
+ * ******************************************************
+ */
+
+public class OverfittingGermanCreditClassification {
+
+    private static int labelIndex = 63; // index of the column of the labels/classes
+    private static int numOfClasses = 2; // number of classification labels/classes
+    private static int numOfFeatures = 63; // number of features to be fed to the model
+//    private static int batchSize = 128; // batch size
+    private static int numOfEpochs = 10000; // number of Epochs for model training
+    private static int numOfHiddenNodes = 50; // number of hidden nodes for classification model
+    private static INDArray weightsArray = Nd4j.create(new double[]{1.0, 0.43}); // weights array for weighted loss function for imbalance data.
+
+    public static void main(String[] args) throws Exception{
+        // 1. ======== load data ========
+        RecordReader data = load_data("/germanCredit/germanCredit.csv"); // call the function to load data
+
+
+        // 2. ======== transform data ========
+        List<List<Writable>> transformedData = transform_data(data); // call the function to transform data
+
+
+        // 3. ======== shuffle data ========
+        DataSet dataset = shuffle_data(transformedData); // call the function to shuffle data
+
+
+        // 4. ======== split train/test set with 70:30 ratio (train:test) and create iterators ========
+        SplitTestAndTrain testAndTrain = dataset.splitTestAndTrain(0.7); // use 70% of the whole data for training set and 30% for test set
+
+        DataSet trainSet = testAndTrain.getTrain();
+        DataSet testSet = testAndTrain.getTest();
+
+        // create data iterators for train set set and test set
+        DataSetIterator trainIterator = new ViewIterator(trainSet, trainSet.numExamples());
+        DataSetIterator testIterator = new ViewIterator(testSet, testSet.numExamples());
+
+
+        // 5. ======== build and initialise the model ========
+        MultiLayerNetwork model = build_model(weightsArray); // call the function to build the model
+        model.init();
+
+
+        // 6. ======== configure listener (UI & Training Loss Value) ========
+        UIServer uiServer = UIServer.getInstance();
+        StatsStorage statsStorage = new FileStatsStorage(new File(System.getProperty("java.io.tmpdir"), "ui-stats.dl4j"));
+        uiServer.attach(statsStorage);
+        model.setListeners(
+                new StatsListener( statsStorage),
+                new ScoreIterationListener(10)
+        );
+
+//        //  ======== Uncomment this section to use Early Stopping ========
+//        // 7. ======== fit the model ========
+//        // **Solution 1**: early stopping - train the model with optimal number of Epochs from performing early stopping
+//        EarlyStoppingConfiguration esConfig = new EarlyStoppingConfiguration.Builder()
```

Comment:
Might not be necessary to include this in training file so as to train participants to code this out.